File: backend_analisis_exergetico/schemas/sustancia_en_flujo.py
from fastapi_utils.api_model import APIModel

from schemas.sustancia import Sustancia


class SustanciaEnFlujoBase(APIModel):
    id_flujo: int
    id_sustancia: int | None


class SustanciaEnFlujoCreate(SustanciaEnFlujoBase):
    pass


class SustanciaEnFlujoUpdate(SustanciaEnFlujoBase):
    pass


class SustanciaEnFlujo(SustanciaEnFlujoBase):
    id: int
    sustancia: Sustancia | None

File: frontend_analisis_exergetico/src/modules/exergia/models/sustancia-en-flujo.ts
import { BaseModel } from 'src/utils/types';
import { Sustancia } from './sustancia';

export interface SustanciaEnFlujoBase extends BaseModel {
  idFlujo: number;
  idSustancia?: number;
}

export type SustanciaEnFlujoCreateBase = SustanciaEnFlujoBase;
export type SustanciaEnFlujoUpdateBase = SustanciaEnFlujoBase;

export interface SustanciaEnFlujo extends SustanciaEnFlujoBase {
  id: number;
  sustancia?: Sustancia;
}
